dcp-0007/dcp-0007.mediawiki

```diff
@@ -0,0 +1,193 @@
+
+For mainnet, the expenditure policy window has a size of 6912 blocks and 6
+preceding windows are checked, for a total of 41472 blocks. A bootstrap amount
+of 16K DCR is used.
+
+A deeper overview of the issue can be read on the 
+[[https://blog.decred.org/2021/06/25/Treasury-Expenditure-Policy-Bug/|Treasury Expenditure Policy Bug]]
+blog post.
+
+==Specification==
+
+Change the maximum expenditure policy to the following:
+
+<blockquote>The sum of treasury spend amounts inside the most recent expenditure window
+cannot exceed the total amount received by the treasury in the same window, in
+addition to a 50% increase.</blockquote>
+
+==Rationale==
+
+Changing the expenditure policy to the one originally specified and approved by
+stakeholders in Politeia should not be controversial from a community discussion
+standpoint and solves the immediate issues facing the blocking of the funds by
+the treasury.
+
+The two policies are largely compatible at this point time, given the current
+average expenditure amount being withdrawn from the treasury, as well as the
+currently mined test transaction in mainnet being acceptable by both policies.
+
+Therefore, if no treasury spend transactions are mined such that they are
+incompatible to the new rule until the new rule is activated and sufficiently
+confirmed, it will be possible to remove legacy code dealing with the old
+policy.
+
+==Deployment==
+
+This agenda will be deployed to mainnet using the standard Decred on-chain
+voting infrastructure as follows:
+
+{|
+!Name!!Setting
+|-
+|Deployment Version||9
+|-
+|Agenda ID||treasuryexppolicy
+|-
+|Agenda Description||Modify the treasury maximum expenditure policy as defined
+in DCP0006
+|-
+|Start Time||XXXXX
+|-
+|Expire Time||XXXXX
+|-
+|Mask||0x06 (Bits 1 and 2)
+|-
+|Choices||{| !Choice!!English Description!!Bits
+|-
+|abstain||abstain voting for change||0x00
+|-
+|no||keep the existing consensus rules||0x02 (Bit 1)
+|-
+|yes||change to the new consensus rules||0x04 (Bit 2)
+|}
+|}
+
+==Compatibility==
+
+This is a hard-forking change to the Decred consensus.  This means that once the
+agenda is voted in and becomes locked in, anybody running code that fully
+validates blocks must upgrade before the activation time or they will risk
+rejecting a chain containing a transaction that is invalid under the old rules.
+
+Other software that performs full validation will need to modify their consensus
+enforcement rules accordingly.
+
+The changes introduced by this proposal modify block acceptance criteria, but no
+transaction semantics are affected, therefore software other than full nodes or
+ones dealing directly with accounting of treasury transactions will not be
+affected by these changes.
+
+==Reference Implementation==
+
+<pre>
+policyWindow := b.chainParams.TreasuryVoteInterval *
+    b.chainParams.TreasuryVoteIntervalMultiplier *
+    b.chainParams.TreasuryExpenditureWindow
+
+// First: sum up tspends, tadds and tbases inside the most recent
```

Comment:
I would suggest calling these out more explicitly.  I personally find the "tfoo" convention ambiguous for people not already familiar and certainly nothing else up to this point refers to `treasurybases` a `tbases`.

e.g. treasury spends, treasury adds, and treasurybases.  The difference in the spacing is so that treasurybase lines up with coinbase.